hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/TransactionInfoRepair.java

```diff
@@ -0,0 +1,128 @@
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.TRANSACTION_INFO_TABLE;
+
+
+/**
+ * Tool to update the highest term-index in transactionInfoTable.
+ */
+@CommandLine.Command(
+    name = "transaction",
+    description = "CLI to update the highest index in transactionInfoTable.",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+@MetaInfServices(SubcommandWithParent.class)
+public class TransactionInfoRepair
+    implements Callable<Void>, SubcommandWithParent  {
+
+  protected static final Logger LOG = LoggerFactory.getLogger(TransactionInfoRepair.class);
+
+  @CommandLine.Spec
+  private static CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.ParentCommand
+  private RDBRepair parent;
+
+  @CommandLine.Option(names = {"--highest-transaction"},
+      required = true,
+      description = "Highest termIndex of transactionInfoTable. The input format is: {term}#{index}.")
+  private String highestTransactionTermIndex;
+
+
+  protected void setHighestTransactionTermIndex(
+      String highestTransactionTermIndex) {
+    this.highestTransactionTermIndex = highestTransactionTermIndex;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    List<ColumnFamilyHandle> cfHandleList = new ArrayList<>();
+    String dbPath = getParent().getDbPath();
+    List<ColumnFamilyDescriptor> cfDescList = RocksDBUtils.getColumnFamilyDescriptors(
+        dbPath);
+
+    try (ManagedRocksDB db = ManagedRocksDB.open(dbPath, cfDescList, cfHandleList)) {
+      ColumnFamilyHandle transactionInfoCfh = RocksDBUtils.getColumnFamilyHandle(TRANSACTION_INFO_TABLE, cfHandleList);
+      if (transactionInfoCfh == null) {
+        System.err.println(TRANSACTION_INFO_TABLE + " is not in a column family in DB for the given path.");
+        return null;
+      }
+      TransactionInfo originalTransactionInfo =
+          RocksDBUtils.getValue(db, transactionInfoCfh, TRANSACTION_INFO_KEY, TransactionInfo.getCodec());
+
+      System.out.println("The original highest transaction Info was " + originalTransactionInfo.getTermIndex());
+
+      TransactionInfo transactionInfo = TransactionInfo.valueOf(
+          highestTransactionTermIndex);
+
+      byte[] transactionInfoBytes = TransactionInfo.getCodec().toPersistedFormat(transactionInfo);
+      db.get()
+          .put(transactionInfoCfh, StringCodec.get().toPersistedFormat(TRANSACTION_INFO_KEY), transactionInfoBytes);
+
+      System.out.println("The highest transaction info has been updated to: " +
+          RocksDBUtils.getValue(db, transactionInfoCfh, TRANSACTION_INFO_KEY,
+              TransactionInfo.getCodec()).getTermIndex());
+    } catch (RocksDBException exception) {
+      System.err.println("Failed to update the RocksDB for the given path: " + dbPath);
+      System.err.println(
+          "Make sure that Ozone entity (OM, SCM or DN) is not running for the give database path and current host.");
```

Comment:
Referencing two earlier comments here:
- We should be be specific about which entity this is currently supported on. I'm pretty sure this should just be OM right now.
- We need to make sure the command exits non-zero in this case.

Reply:
makes sense! added description for OM right now, and throw exception after printing the error message.